### PR TITLE
Implement PerformanceNavigation interface

### LINKED ISSUE #22828: Implement deprecated PerformanceNavigation interface
Spec:
https://w3c.github.io/navigation-timing/#the-performancenavigation-interface
https://w3c.github.io/navigation-timing/#extensions-to-the-performance-interface

Code: `components/script/dom/performance.rs`, `components/script/dom/webidls/performance.webidl`

This will require [adding a new DOM interface](https://doc.servo.org/script/dom/index.html#adding-a-new-dom-interface) for PerformanceNavigation.

http://software.hixie.ch/utilities/js/live-dom-viewer/saved/6745
This page should show "log: 0" in the log table when the interface exists.

Loading twitter in Servo shows a JS error where it assumes this interface exists. Adding this implementation should make that error go away and make twitter behave better in Servo.

### PR DESCRIPTION
**Pull request by iulianR**
This is my first attempt. Please let me know if I should change anything.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #22828 (GitHub issue number if applicable)

- [ ] There are tests for these changes OR
- [x] These changes do not require tests because **not sure**

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23203)

### CODE REVIEW

**Comment by iulianR** [comment]
I see that I should update the `mozzila/interfaces.html` with `PerformanceNavigation`, but I'm not sure what to do with the tests in `worker-performance.worker.js`. What should they test now?

File: components/script/dom/webidls/Performance.webidl [hunk 1]
```diff
@@ -49,3 +49,11 @@ partial interface Performance {
 partial interface Performance {
   PerformanceNavigationTiming timing();
 };
+// https://w3c.github.io/navigation-timing/#extensions-to-the-performance-interface
+[Exposed=Window]
```
Comment (jdm) [inline components/script/dom/webidls/Performance.webidl]: What if you move this Exposed annotation to the attribute instead, like:
```idl
partial interface Performance {
  [SameObject, Exposed=Window]
  readonly attribute PerformanceNavigation navigation;
}
```
If you run `./mach test-wpt tests/wpt/web-platform-tests/workers/worker-performance.worker.html` then there should not be any unexpected results.

**Comment by jdm** [comment]
There are also a lot of new test results in https://build.servo.org/builders/linux-rel-wpt/builds/12027/steps/shell__4/logs/filtered-wpt-errorsummary.log . I recommend following the instructions at https://github.com/servo/servo/blob/master/tests/wpt/README.md#updating-test-expectations using a log generated from `./mach test-wpt tests/wpt/web-platform-tests/navigation-timing/ --log-raw /tmp/servo.log`.

**Comment by iulianR** [comment]
I also noticed this warning that I don't think was there before. It is, however, not clear to me what it means.
```
warning: unused attribute
  --> components/script/dom/bindings/conversions.rs:74:5
   |
74 |     rustc_on_unimplemented(message = "The IDL interface `{Self}` is not derived from `{T}`.")
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: #[warn(unused_attributes)] on by default
```

**Comment by jdm** [comment]
It comes and goes, and is not related to your changes.

**Comment by iulianR** [comment]
I updated the tests as you asked, although there were more changes than I expected.

**Comment by jdm** [comment]
@iulianR Would you mind [squashing](https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit) your commits into one?

**Comment by jdm** [comment]
@bors-servo r+
Sorry about the delay!

**Comment by jdm** [comment]
Huh. It's not clear to me why a property that is marked Exposed=Window is accessible from a worker :<

**Comment by jdm** [comment]
I'm digging into the issue with the worker separately. I recommend updating the expected test results for /workers/worker-performance.worker.html as an expected failure so we can merge these changes.

**Comment by jdm** [comment]
@iulianR Are you planning to finish this s work by updating the expected test result?

**Comment by iulianR** [comment]
Yes, I will finish it. I forgot about it.

**Comment by iulianR** [comment]
Hmm, I assumed the problem with the worker was fixed now, so I didn't update the expected test result.

> I'm digging into the issue with the worker separately. I recommend updating the expected test results for /workers/worker-performance.worker.html as an expected failure so we can merge these changes.
I will do this now.

**Comment by jdm** [comment]
The worker issue is being fixed in #23500 but isn't ready yet.

**Comment by jdm** [comment]
@iulianR Are you still planning to finish this? :)

**Comment by iulianR** [comment]
@jdm Can you tell me how I should run the tests so I can figure out if the change I made (locally) was good? I still don't fully understand the testing framework.

**Comment by jdm** [comment]
You can run tests locally like so: `./mach test-wpt tests/wpt/web-platform-tests/workers/worker-performance.any.js`

**Comment by jdm** [comment]
Since #23500 has merged, let's see if that's good enough, though!
@bors-servo try=wpt

**Pull request merged**